skt/reporter.py

```diff
@@ -117,30 +117,36 @@ def __init__(self, kver, url):
         """
```

Comment:
Commenting in the code, but this should be a comment to the commit as a whole, which GitHub doesn't allow at this moment, unfortunately.

This commit has several logical changes, as far as I can see:
* Fixing handling kernels not booting at all
* Switching to explicit fetching, upon creation (along with test update)
* Handling the missing URL
* Refactoring line extraction

Those would be better as separate commits for simpler reverting and cherry-picking.

Reply:
> - Fixing handling kernels not booting at all

It did the same thing before, I just added the comment to explicitly state it to avoid confusion.

> - Handling the missing URL

The missing URL was handled on the reporter side before, I just moved it here because it makes more sense (and in the followup commit, removed the URL check from reporter so it doesn't break anything).

I didn't really want to separate fetching upon extraction and it's refactoring since they are related and I wanted to make sure each added commit works separately.

Reply:
I agree, each commit should work separately, thank you for caring about that. Pity none of the CI solutions available on GitHub support verifying every commit in a PR, only the last.

I would like to understand your intention for the way you split the changes into commits better, and share my view on this, but I'm afraid I'll stray too far and delay us. This looks already better to me and I think it's good for the start. Let's return to the topic in the next PR :)